src/classes/OPP_OpportunityContactRoles_TDTM.cls

```diff
@@ -495,21 +537,50 @@ public without sharing class OPP_OpportunityContactRoles_TDTM extends TDTM_Runna
     ********************************************************************************************************/
     private boolean needsManageOCR(Opportunity opp, Opportunity oldOpp, string fieldName) {
         //no role for this field
-        if (string.isBlank(getRole(fieldName)))
+        if (String.isBlank(getRole(fieldName, null)))
             return false;
         //field value hasn't changed
         if (opp.get(fieldName) == oldOpp.get(fieldName))
             return false;
         //the honoree contact is the same as the primary contact, thus is already managed
-        if (fieldName == 'Honoree_Contact__c' && opp.Honoree_Contact__c == opp.Primary_Contact__c)
+        if (fieldName == FIELD_NAME_HONOREE_CONTACT && opp.Honoree_Contact__c == opp.Primary_Contact__c)
             return false;
         //the notification recipient contact is the same as the primary contact, thus the OCR is already managed
-        if (fieldName == 'Notification_Recipient_Contact__c' && opp.Notification_Recipient_Contact__c == opp.Primary_Contact__c)
+        if (fieldName == FIELD_NAME_NOTIFICATION_RECIPENT_CONTACT && opp.Notification_Recipient_Contact__c == opp.Primary_Contact__c)
             return false;
         //the notification recipient contact is the same as the honoree contact, thus the OCR is already managed
-        if (fieldName == 'Notification_Recipient_Contact__c' && opp.Notification_Recipient_Contact__c == opp.Honoree_Contact__c)
+        if (fieldName == FIELD_NAME_NOTIFICATION_RECIPENT_CONTACT && opp.Notification_Recipient_Contact__c == opp.Honoree_Contact__c)
             return false;
 
         return true;
     }
+
+
+    /*********************************************************************************************************
+    * @description Retrieves details/information about the Opportunity's associated Account.
+    * @param currentOpportunities The Opportunities in the 'new' trigger context variable.
+    * @return Map<Id, Account> containing the AccountId to Account.
+    **********************************************************************************************************/
+    public static  Map<Id, Account> populateAccountDetails (List<Opportunity> currentOpportunities) {
+        Map<Id, Account> accountDetails = null;
+        List<Id> accountIds = new List<Id>();
+
+        if (currentOpportunities.isEmpty()) {
+            return accountDetails;
+        }
+
+        for (Opportunity eachOppty : currentOpportunities) {
+            if (eachOppty.AccountId != null) {
+                accountIds.add(eachOppty.AccountId);
+            }
+        }
+
+        accountDetails = new Map<Id, Account>([SELECT Id, Name, RecordTypeId, npe01__SYSTEMIsIndividual__c
```

Comment:
as mentioned earlier, let's see if this method and extra query is really necessary.

Reply:
The query is necessary.